dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/RecoverInterruptedJob.java

```diff
@@ -54,168 +65,180 @@
+        } catch (final IOException e) {
+            LOG.error("IO", e);
+
+        } catch (final InterruptedException e) {
+            LOG.error("Interuption", e);
+
+        }
+        loggingService.logMessage(buildFinalMessage, SUCCESS);
+        removeJobIdAndUpdateJobsBtn(jobInterruptionStore, uuid);
+
+    }
+
+    private void setDataTransferredListener(final long l, final Long totalJobSize, final AtomicLong totalSent) {
+        updateProgress(totalSent.getAndAdd(l) / 2, totalJobSize);
+        totalSent.addAndGet(l);
+    }
+
+    private static void waitForTransfer(final FilesAndFolderMap filesAndFolderMap, final UUID jobId, final boolean isCacheJobEnable, final Ds3Client ds3Client) throws IOException, InterruptedException {
+        final GetJobSpectraS3Response response = ds3Client.getJobSpectraS3(new GetJobSpectraS3Request(jobId));
+        if (cacheIsEnabledAndIsPut(filesAndFolderMap, isCacheJobEnable)) {
+            retryJob(response, ds3Client, jobId);
         }
     }
 
-    private void getJob(final FilesAndFolderMap filesAndFolderMap) throws Exception {
-        final Ds3ClientHelpers helpers = Ds3ClientHelpers.wrap(ds3Client, 100);
-        if (filesAndFolderMap.getType().equals(PUT.toString())) {
-            job = helpers.recoverWriteJob(uuid);
-            updateMessage(StringBuilderUtil.getRecoverJobInitiateTransferTo(job.getBucketName()).toString());
+    private void updateProgressAndLog(final JobRequestType jobRequestType, final long totalJobSize, final AtomicLong totalSent, final String buildGetRecoveringMessage, final String buildPutRecoveringMessage) {
+        if (jobRequestType == GET) {
+            updateMessage(buildGetRecoveringMessage);
+            loggingService.logMessage(buildGetRecoveringMessage, SUCCESS);
+        } else {
+            updateMessage(buildPutRecoveringMessage);
+            loggingService.logMessage(buildPutRecoveringMessage, SUCCESS);
+        }
+        updateProgress(totalSent.get(), totalJobSize);
+    }
 
-        } else if (filesAndFolderMap.getType().equals(JobRequestType.GET.toString())) {
-            job = helpers.recoverReadJob(uuid);
-            updateMessage(StringBuilderUtil.getRecoverJobInitiateTransferFrom(job.getBucketName()).toString());
+    private static String buildFinalMessage(final String targetLocation, final long totalJobSize, final ResourceBundle resourceBundle) {
+        return resourceBundle.getString("jobSize")
+                + StringConstants.SPACE + FileSizeFormat.getFileSizeType(totalJobSize) + StringConstants.SPACE
+                + resourceBundle.getString("recoveryCompleted") + StringConstants.SPACE
+                + targetLocation + StringConstants.SPACE
+                + resourceBundle.getString("storageLocation");
+    }
+
+    private static void retryJob(final GetJobSpectraS3Response response, final Ds3Client ds3Client, final UUID jobId) throws InterruptedException, IOException {
```

Comment:
It would be interesting to see how this would look using RxJava's retry/delay mechanism.

Reply:
Ds3PutJob @ line 319 does something similar with RxJava